src/HieDb/Create.hs

```diff
@@ -214,8 +214,25 @@ addRefsFromLoaded
   -> Fingerprint -- ^ The hash of the @.hie@ file
   -> HieFile -- ^ Data loaded from the @.hie@ file
   -> m ()
-addRefsFromLoaded db@(getConn -> conn) path sourceFile hash hf = liftIO $ withTransaction conn $ do
-  deleteInternalTables conn path
+addRefsFromLoaded
+  db@(getConn -> conn) path sourceFile hash hf =
+    liftIO $ withTransaction conn $ do
+      deleteInternalTables conn path
+      addRefsFromLoaded_unsafe db path sourceFile hash hf
+
+-- | Like 'addRefsFromLoaded' but without using a transaction
+addRefsFromLoaded_unsafe
+  :: MonadIO m
+  => HieDb -- ^ HieDb into which we're adding the file
+  -> FilePath -- ^ Path to @.hie@ file
+  -> SourceFile -- ^ Path to .hs file from which @.hie@ file was created
+                -- Also tells us if this is a real source file?
+                -- i.e. does it come from user's project (as opposed to from project's dependency)?
+  -> Fingerprint -- ^ The hash of the @.hie@ file
+  -> HieFile -- ^ Data loaded from the @.hie@ file
+  -> m ()
+addRefsFromLoaded_unsafe
+ db@(getConn -> conn) path sourceFile hash hf = liftIO $ do
```

Comment:
You still have to call `deleteInternalTables` I think.

Reply:
Why is it needed on every individual add?

Reply:
Note that it is called by the "safe" version of `addRefs`, it' just the unsafe version that does not call it implicitly

Reply:
We want to delete references from the previous version of the file before we insert new references.

Reply:
I'd still rather keep it out from the unsafe version though - since I am using it to create a new database so I know that there are no previous versions and I want it to be as fast as possible

Reply:
What do you say about noting this in the haddocks?

Reply:
comments added!